pkg/ebpf/events_processor.go

```diff
@@ -131,20 +140,13 @@ func (t *Tracee) processEvent(event *external.Event) error {
 		}
 
 	case SchedProcessExecEventID:
-		//update the process tree
-		processData := procinfo.ProcessCtx{
-			StartTime:   event.Timestamp,
-			ContainerID: event.ContainerID,
-			Pid:         uint32(event.ProcessID),
-			Tid:         uint32(event.ThreadID),
-			Ppid:        uint32(event.ParentProcessID),
-			HostTid:     uint32(event.HostThreadID),
-			HostPid:     uint32(event.HostProcessID),
-			HostPpid:    uint32(event.HostParentProcessID),
-			Uid:         uint32(event.UserID),
-			MntId:       uint32(event.MountNS),
-			PidId:       uint32(event.PIDNS)}
-		t.procInfo.UpdateElement(event.HostThreadID, processData)
+		//update the process tree with correct comm name
+		processData, err := t.procInfo.GetElement(event.HostThreadID)
```

Comment:
For clarity, I think we should use HostProcessID instead (pid should be equal to tid at that point)

Reply:
yes they should be equal at that point. but the goal is to update comm for each thread in the process tree. every operation on the process tree is done on the context of the current thread (not process). don't you think?

Reply:
If you only update comm in exec, it doesn't matter if you take thread or process, as they are equal at that point.
To update comm in thread resolution we should also monitor prctl event with command to change thread name

Reply:
ok, i'll use pid for clarity